data/src/main/java/com/github/leuludyha/data/db/LibraryDao.kt

```diff
@@ -128,4 +133,71 @@ interface LibraryDao {
     @Transaction
     @Query("SELECT * FROM editions WHERE editionId LIKE :editionId")
     fun getEditionWithCovers(editionId: String): Flow<EditionWithCovers>
+
+    suspend fun insert(work: Work) {
+        val workEntity = WorkEntity.from(work)
+
+        val authors = work.authors.firstOrNull()
+        authors?.forEach { insert(it) }
+        val workAuthorCrossRefs = authors?.map { WorkAuthorCrossRef.from(work, it) }
+
+        val editions = work.editions.firstOrNull()
+        editions?.forEach { insert(it) }
+        val workEditionCrossRefs = editions?.map { WorkEditionCrossRef.from(work, it) }
+
+        val subjects = work.subjects.firstOrNull()
+        subjects?.forEach { insertSubject(it) }
```

Comment:
Why does `insertSubject()` has its method name different from the convention you used for other inserts ? Is it because it has type `String` ?

Reply:
Yes it is, to make it clear it is supposed to take a subject as argument
Many times I was hesitating to create a
```kotlin
data class Subject(subject: String)
```
Do you think we should?